src/AutorestSwift/View Models/OperationViewModel.swift

```diff
@@ -102,30 +103,33 @@ struct OperationViewModel {
             operation: operation
         )
 
-        var signatureParams = filterParams(for: operation.signatureParameters, with: [.path, .uri, .body])
+        var signatureParams = filterParams(for: operation.signatureParameters, with: [.path, .uri])
         var optionsParams = filterParams(for: operation.signatureParameters, with: [.header, .query])
 
         var requests = [RequestViewModel]()
         var responses = [ResponseViewModel]()
 
-        for request in operation.requests ?? [] {
-            requests.append(RequestViewModel(from: request))
+        assert(
+            operation.requests?.count ?? 0 <= 1,
+            "Multiple requests per operation is currently not supported... \(operation.name)"
+        )
+        guard let request = operation.requests?.first else { fatalError("No requests found.") }
+        requests.append(RequestViewModel(from: request, with: operation))
 
-            let requestSignatureParams = filterParams(for: request.signatureParameters, with: [.path, .uri, .body])
-            let requestOptionsParams = filterParams(
-                for: request.signatureParameters,
-                with: [.header, .query]
-            )
+        let requestSignatureParams = filterParams(for: request.signatureParameters, with: [.path, .uri])
```

Comment:
so you remove . body from the requestSignaureParams since it is handled specially (put that as first of the signature)

Reply:
Exactly. I think swift arrays are ordered, but this guarantees we can treat any body params first.